src/apiNetworkProvider.ts

```diff
@@ -16,17 +16,23 @@ import { DefinitionOfFungibleTokenOnNetwork, DefinitionOfTokenCollectionOnNetwor
 import { FungibleTokenOfAccountOnNetwork, NonFungibleTokenOfAccountOnNetwork } from "./tokens";
 import { TransactionOnNetwork, prepareTransactionForBroadcasting } from "./transactions";
 import { TransactionStatus } from "./transactionStatus";
+import { extendUserAgent } from "./userAgent";
+import { NetworkProviderConfig } from "./networkProviderConfig";
+import { MetricsPrefix } from "./constants";
 
 // TODO: Find & remove duplicate code between "ProxyNetworkProvider" and "ApiNetworkProvider".
 export class ApiNetworkProvider implements INetworkProvider {
     private url: string;
-    private config: AxiosRequestConfig;
+    private config: NetworkProviderConfig;
     private backingProxyNetworkProvider;
+    private userAgentPrefix = `${MetricsPrefix}/api`
 
-    constructor(url: string, config?: AxiosRequestConfig) {
+    constructor(url: string, config?: NetworkProviderConfig) {
         this.url = url;
         this.config = { ...defaultAxiosConfig, ...config };
-        this.backingProxyNetworkProvider = new ProxyNetworkProvider(url, config);
+        const proxyConfig = JSON.parse(JSON.stringify(this.config));
```

Comment:
How about `structuredClone()`, does it help?

Reply:
in order to use this I need to upgrade the types/node to a version higher then 17, maybe ww will do that with a major release